eclair-core/src/main/scala/fr/acinq/eclair/io/PeerConnection.scala

```diff
@@ -358,6 +358,15 @@ class PeerConnection(keyPair: KeyPair, conf: PeerConnection.Conf, switchboard: A
         }
         stay using d.copy(behavior = behavior1)
 
+      case Event(DoSync, d: ConnectedData) =>
+        val flags_opt = if (d.localInit.features.hasFeature(Features.ChannelRangeQueriesExtended) && d.remoteInit.features.hasFeature(Features.ChannelRangeQueriesExtended)) {
+          Some(QueryChannelRangeTlv.QueryFlags(QueryChannelRangeTlv.QueryFlags.WANT_ALL))
+        } else {
+          None
+        }
+        router ! SendChannelQuery(d.chainHash, d.remoteNodeId, self, replacePrevious = false, flags_opt)
+        stay
+
```

Comment:
What if our peer does not support channel queries at all (i.e they don't even set `Features.ChannelRangeQueries`) ?

Reply:
Good question, I'll dig into it.

Reply:
Fixed in https://github.com/ACINQ/eclair/pull/1587/commits/e5ba331c3cdc0984c999126ecf4eacd5ea53a701